django_admin_listfilter_dropdown/templates/admin/change_list.html

```diff
@@ -0,0 +1,52 @@
+{% extends "admin/change_list.html" %}
+{% load staticfiles %}
+
+{% block extrahead %}
+    <script type="text/javascript" src="{% static 'admin/js/vendor/jquery/jquery.js' %}"></script>
+    <script type="text/javascript" src="{% static 'admin/js/vendor/select2/select2.full.js' %}"></script>
+
+    <script>
+        (function () {
+            var init = function ($element, options) {
+                var settings = $.extend({
+                    ajax: {
+                        data: function (params) {
+                            return {
+                                term: params.term,
+                                page: params.page
+                            };
+                        }
+                    }
+                }, options);
+                $element.select2(settings).on('change', function (ev) {
+                    var filter_param = jQuery(this).attr('name') + '__id__exact';
+                    // TODO: Not all browsers support URLSearchParams
+                    // https://caniuse.com/#search=URLSearchParams
+                    params = new URLSearchParams(window.location.search);
```

Comment:
browser coverage is not bad, but IE and the odd mobile browsers dont support it (yet). Unfortunately jQuery does not have tools to handle url search params :( - any elegant solutions?